wasm/tests/offchain.rs

```diff
@@ -315,3 +731,73 @@ async fn test_import_resolution() {
     assert_eq!(outputs.len(), 1);
     assert_eq!(outputs[0], "30u32");
 }
+
+#[wasm_bindgen_test]
+async fn test_fee_calculation_v1() {
+    // Transaction id: at1fy95sxgr267e6vghzzg6effs83cm07lkgufzentkvx9f0pqldsxs67zhrk
+    // Block number: 456,789 on testnet
+    let private_key = PrivateKey::from_string("APrivateKey1zkp3dQx4WASWYQVWKkq14v3RoQDfY2kbLssUj7iifi1VUQ6").unwrap();
+    let expected_microcredits = 1_449_u64; // value from the block explorer
+    let input_1 = "aleo18clqv2ycpdmz07mzsuevp6yqwcz6ym303pra9hly523rpjuw0y9q5anej2";
+    let input_2 = "1000000u64";
+    let inputs = Array::new();
+    inputs.set(0u32, JsValue::from_str(input_1));
+    inputs.set(1u32, JsValue::from_str(input_2));
+
+    let offline_query = OfflineQuery::from_string(OFFLINE_QUERY_V1).unwrap();
+
+    let transaction = ProgramManager::execute(
+        &private_key,
+        PUZZLE_ARCADE_COIN_V001_PROGRAM,
+        "mint",
+        inputs,
+        0.0,
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+        Some(offline_query),
+    )
+    .await;
+
+    let transaction_fee = transaction.unwrap().fee_amount();
+    assert_eq!(transaction_fee, expected_microcredits);
+}
+
+#[wasm_bindgen_test]
+async fn test_fee_calculation_v2() {
+    // Trasaction ID:  at1av6d606xj04w4fmqlp088z0m0jnwqxr00mz2qkcap2xyxeu6ngxs7f3gx7
+    // Block number: 6,398,077 on testnet
+    let private_key = PrivateKey::from_string("APrivateKey1zkp3dQx4WASWYQVWKkq14v3RoQDfY2kbLssUj7iifi1VUQ6").unwrap();
+    let expected_microcredits = 17_939_u64;
+    let input_1 = "aleo1lly6xyqmg9y2xgug9wd8s5suvgpp0wp27zjtyfgk9p0w996v3yxsg90zk5";
+    let input_2 = "14973561u64";
+    let inputs = Array::new();
+    inputs.set(0u32, JsValue::from_str(input_1));
+    inputs.set(1u32, JsValue::from_str(input_2));
+
+    let offline_query = OfflineQuery::from_string(OFFLINE_QUERY_V2).unwrap();
+
+    let transaction = ProgramManager::execute(
+        &private_key,
+        CREDITS_ALEO_PROGRAM,
```

Comment:
You can simply change it here to this.

```suggestion
        Program::credits().to_string(),
```

Reply:
Done:  [e453562](https://github.com/ProvableHQ/sdk/pull/1007/commits/e453562485e0590a524d1484b217641a559f977a)